typings/number_format.d.ts

```diff
@@ -0,0 +1,54 @@
+/// <reference types="react" />
+
+declare module "react-number-format" {
+  export interface NumberFormatState {
+    value?: string;
+    numAsString?: string;
+  }
+
+  export interface NumberFormatValues {
+    floatValue: number;
+    formattedValue: string;
+    value: string;
+  }
+
+  type FormatValueFunction = (formattedValue: string) => string;
+
+  export interface NumberFormatProps
+    extends React.HTMLAttributes<HTMLInputElement> {
+    thousandSeparator?: boolean | "," | "." | " ";
+    decimalSeparator?: boolean | "," | ".";
+    decimalScale?: number;
+    fixedDecimalScale?: boolean;
+    displayType?: "input" | "text";
+    prefix?: string;
+    suffix?: string;
+    format?: string | FormatValueFunction;
+    removeFormatting?: (formattedValue: string) => string;
+    mask?: string | string[];
+    value?: number | string;
+    isNumericString?: boolean;
+    customInput?: (inputComponent: React.ComponentType<any>) => void;
+    allowNegative?: boolean;
+    allowEmptyFormatting?: boolean;
+    onValueChange?: (
+      values: NumberFormatValues,
+      e: React.ChangeEvent<HTMLInputElement>

```

Comment:
e will not be always a ChangeEvent, It can be blur event as well. Some transformation happens on blur as well which calls onValueChange

Reply:
Should I set e to any or something similar to SyntheticInputEvent?

Reply:
SyntheticInputEvent will be better. If its not available you can change it to something like `e: React.ChangeEvent<HTMLInputElement> | React.BlurEvent<HTMLInputElement>`